File: easy/go/func_call.go
package main

import (
	"vulndemo/easy/go/cmd_exec"
	"vulndemo/easy/go/common"
	"vulndemo/easy/go/iface"
)

func RunByChild(cmd string) string {
	child := common.VulnChild{}
	return child.Run(cmd)
}

func RunByInterface(cmd string) string {
	var runner iface.VulnRunner = cmd_exec.InterfaceImpl{}
	return runner.Run(cmd)
}

File: easy/rust/func_call.rs
use crate::common::vuln_child::VulnChild;
use crate::common::vuln_parent::VulnParent;
use crate::cmd_exec::interface_impl::InterfaceImpl;
use crate::iface::vuln_runner::VulnRunner;

pub fn run_by_child(cmd: &str) {
    let child = VulnChild { parent: VulnParent };
    child.parent.exec(cmd);
}

pub fn run_by_interface(cmd: &str) {
    let impl_obj = InterfaceImpl {
        child: VulnChild { parent: VulnParent },
    };
    let runner: &dyn VulnRunner = &impl_obj;
    runner.run(cmd);
}
